Jellyfin.Server.Implementations/Users/UserManager.cs

```diff
@@ -0,0 +1,824 @@
+        /// <inheritdoc/>
+        public User? GetUserById(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                throw new ArgumentException("Guid can't be empty", nameof(id));
+            }
+
+            return _dbProvider.CreateContext().Users.Find(id);
+        }
+
+        /// <inheritdoc/>
+        public User? GetUserByName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("Invalid username", nameof(name));
+            }
+
+            // This can't use an overload with StringComparer because that would cause the query to
+            // have to be evaluated client-side.
+            return _dbProvider.CreateContext().Users.FirstOrDefault(u => string.Equals(u.Username, name));
+        }
+
+        /// <inheritdoc/>
+        public async Task RenameUser(User user, string newName)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            if (string.IsNullOrWhiteSpace(newName))
+            {
+                throw new ArgumentException("Invalid username", nameof(newName));
+            }
+
+            if (user.Username.Equals(newName, StringComparison.Ordinal))
+            {
+                throw new ArgumentException("The new and old names must be different.");
+            }
+
+            if (Users.Any(u => u.Id != user.Id && u.Username.Equals(newName, StringComparison.OrdinalIgnoreCase)))
+            {
+                throw new ArgumentException(string.Format(
+                    CultureInfo.InvariantCulture,
+                    "A user with the name '{0}' already exists.",
+                    newName));
+            }
+
+            user.Username = newName;
+            await UpdateUserAsync(user).ConfigureAwait(false);
+
+            OnUserUpdated?.Invoke(this, new GenericEventArgs<User>(user));
+        }
+
+        /// <inheritdoc/>
+        public void UpdateUser(User user)
+        {
+            var dbContext = _dbProvider.CreateContext();
+            dbContext.Users.Update(user);
+            dbContext.SaveChanges();
+        }
+
+        /// <inheritdoc/>
+        public async Task UpdateUserAsync(User user)
+        {
+            var dbContext = _dbProvider.CreateContext();
+            dbContext.Users.Update(user);
+
+            await dbContext.SaveChangesAsync().ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public User CreateUser(string name)
+        {
+            if (!IsValidUsername(name))
+            {
+                throw new ArgumentException("Usernames can contain unicode symbols, numbers (0-9), dashes (-), underscores (_), apostrophes ('), and periods (.)");
+            }
+
+            var dbContext = _dbProvider.CreateContext();
+
+            // TODO: Remove after user item data is migrated.
+            var max = dbContext.Users.Select(u => u.InternalId).Max();
```

Comment:
```suggestion
            long max;
            if (dbContext.Users.Any())
            {
                max = dbContext.Users.Select(u => u.InternalId).Max();
            }
            else
            {
                max = 0;
            }
```

Max throws an exception if the enumerable is empty. Normally we would use `DefaultIfEmpty(0)` but that also throws an exception.